tfhe/src/shortint/server_key/modulus_switched_compression.rs

```diff
@@ -14,34 +21,48 @@ impl ServerKey {
         &self,
         ct: &Ciphertext,
     ) -> CompressedModulusSwitchedCiphertext {
-        if let ShortintBootstrappingKey::MultiBit { .. } = &self.bootstrapping_key {
-            panic!("Modulus switch compression and multi bit PBS are currently not compatible")
-        }
-
-        let compressed_modulus_switched_lwe_ciphertext = match self.pbs_order {
-            PBSOrder::KeyswitchBootstrap => {
-                let ms_ed = ShortintEngine::with_thread_local_mut(|engine| {
-                    let (mut ciphertext_buffers, _) = engine.get_buffers(self);
-
-                    keyswitch_lwe_ciphertext(
-                        &self.key_switching_key,
-                        &ct.ct,
-                        &mut ciphertext_buffers.buffer_lwe_after_ks,
-                    );
-
-                    CompressedModulusSwitchedLweCiphertext::compress(
-                        &ciphertext_buffers.buffer_lwe_after_ks,
-                        CiphertextModulusLog(self.bootstrapping_key.polynomial_size().log2().0 + 1),
-                    )
-                });
+        let compressed_modulus_switched_lwe_ciphertext =
+            ShortintEngine::with_thread_local_mut(|engine| {
+                let (mut ciphertext_buffers, _) = engine.get_buffers(self);
+                match self.pbs_order {
+                    PBSOrder::KeyswitchBootstrap => {
+                        keyswitch_lwe_ciphertext(
+                            &self.key_switching_key,
+                            &ct.ct,
+                            &mut ciphertext_buffers.buffer_lwe_after_ks,
+                        );
+                    }
+                    PBSOrder::BootstrapKeyswitch => ciphertext_buffers
+                        .buffer_lwe_after_ks
+                        .as_mut_view()
+                        .into_container()
```

Comment:
.as_mut, check for all .into_container usage in your diff please :)

Reply:
Done